cpp/include/cudf/types.hpp

```diff
@@ -136,6 +136,15 @@ enum class nan_policy : bool {
   NAN_IS_VALID  ///< treat nans as valid elements (non-null)
 };
 
+/**
+ * @brief Enum to consider different elements (of floating point types) holding NaN value as equal
+ * or unequal
+ */
+enum class nan_equality : bool {
+  EQUAL,   ///< NaNs compare equal
+  UNEQUAL  ///< NaNs compare unequal
+};
```

Comment:
This is worth more clarification (and perhaps another option). `nan_equality::EQUAL` would make me think `+NaN == +NaN` and `-NaN == -NaN`, but `-NaN != +NaN`. However, I know that you're intending this to mean `+NaN == +NaN`, `+NaN == -NaN`, and `-NaN == -NaN`.

Reply:
How about changing `EQUAL` to `ALWAYS_EQUAL`, or `ALL_EQUAL`? I don't think we ever need an option to have 
```
-NaN == -NaN && NaN == NaN && -NaN != NaN
```

Reply:
How about this
```
enum class nan_equality /*unspecified*/ {
  ALL_EQUAL,   ///< All NaNs compare equal, regardless of sign
  UNEQUAL      ///< All NaNs compare unequal (IEE754 behavior)
};
```

> I think we never need any additional enum option here to have

Given how infuriating both Pandas and Spark are wrt to NaNs, I'm not going to hold my breath on that one.

Reply:
Should we introduce the third behavior now in case someone needs it? I.E. `+NaN == +NaN`, `-NaN == -NaN`, `+NaN != -NaN`?

Reply:
Convince me if there exists someone like that 😄

Reply:
Sorry, I didn't mean to actually implement it in the `collect_set` implementation, but to just add the enum class value. Unfortunately, I don't suspect this will be the last place where we're forced to handle different `NaN` behaviors and we've seen that Spark and Pandas both have semi arbitrary behaviors with this regard.

Reply:
Adding this third behavior (with complete implementation) is very easy here, much easier than the first behavior. I'm just not sure if you guys actually want it/agree to have it. Let's wait a bit to see other opinions (I saw Jake's comment above).

Reply:
We shouldn't add a third option it if we're not going to implement it because that would be super confusing.

Reply:
The third option, `SAME_SIGN_EQUAL`, is online, along with its implementation for `drop_list_duplicates` and corresponding unit tests.

Reply:
I appreciate you added the extra implementation, but I guess what I was implying is that we shouldn't implement things that we don't have a usecase for. That's just unnecessary extra binary size and code to maintain/test.

Reply:
Got it. I reversed my commits 😃